### PR TITLE
Postgres support

### PR DESCRIPTION
**Pull request by jackyzhen**
https://github.com/marianogappa/sql/issues/14

This enables support for querying postgres. It is enabled via setting ```"sqlType": "postgres"``` in `databases.json` config. It is backwards compatible with current configurations as it defaults to mysql if the field is not configured.

### CODE REVIEW

**Comment by jackhopner** [comment]
🎉

File: main.go [hunk 1]
```diff
@@ -78,17 +75,29 @@ func _main(databases map[string]database, databasesArgs []string, query string,
 		targetDatabases = append(targetDatabases, k)
 	}
 
+	sqlTypes := map[sqlType]exists{}
+	var sqlType sqlType
+	for _, db := range targetDatabases {
+		sqlType = databases[db]._sqlType
+		sqlTypes[sqlType] = exists{}
+		if len(sqlTypes) > 1 {
+			usage("More than one sql types specified in target databases.")
```
[the remark this answers is not in the document]
Comment (jackyzhen) [reply main.go]: I just didn't think it made sense or be a likely use case to be hitting different database types with the same query... Its a sign of a bad configuration IMO. But its not hard to support.

File: sql_runner.go [hunk 2]
```diff
@@ -141,6 +141,9 @@ func (sr *sqlRunner) runSQL(db database, key string) bool {
 	scanner := bufio.NewScanner(stdout)
 	for scanner.Scan() {
 		line := scanner.Text()
+		if sr.typ == postgreSQL {
+			line = strings.Replace(line, " | ", "\t", -1)
```
Comment (marianogappa) [inline sql_runner.go]: This is dodgy as :trollface: 
All you need is to add this to the CLI call:

```
-tAF '        '
```

That's a literal tab inside the quotes.

File: main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (marianogappa) [reply main.go]: Many users voted to blindly pass the queries and let users deal with improper yaml configs.

File: sql_runner.go [hunk 3]
```diff
@@ -72,7 +74,12 @@ func mustNewSQLRunner(quitContext context.Context, typ sqlType, printer func(str
 }
 
 func (sr *sqlRunner) runSQL(db database, key string) bool {
-	sqlOptions := sqlTypeToOptions[sr.typ]
+	typ, ok := validSQLTypes[db.SQLType]
+	if !ok {
+		usage("Unknown sql type %v for %v", db.SQLType, db)
```
[the remark this answers is not in the document]
Comment (jackyzhen) [reply sql_runner.go]: Fixed. Hope you don't mind I slightly refactored the error handling too.

**Comment by jackyzhen** [comment]
```
test_1           | 2018/10/25 20:12:03 MySQL ready after 10/30 tries
test_1           | === RUN   Test_MySQL/reads_from_one_database
test_1           | === RUN   Test_MySQL/reads_from_two_databases
test_1           | === RUN   Test_MySQL/reads_from_all_databases_with_the_all_keyword
test_1           | === RUN   Test_MySQL/reads_two_fields_from_all_databases
test_1           | --- PASS: Test_MySQL (9.15s)
test_1           |     --- PASS: Test_MySQL/reads_from_one_database (0.01s)
test_1           |     --- PASS: Test_MySQL/reads_from_two_databases (0.01s)
test_1           |     --- PASS: Test_MySQL/reads_from_all_databases_with_the_all_keyword (0.01s)
test_1           |     --- PASS: Test_MySQL/reads_two_fields_from_all_databases (0.02s)
test_1           | === RUN   Test_PostgreSQL
test_1           | 2018/10/25 20:12:04 PostgreSQL ready after 1/30 tries
test_1           | === RUN   Test_PostgreSQL/reads_from_one_database
test_1           | === RUN   Test_PostgreSQL/reads_from_two_databases
test_1           | === RUN   Test_PostgreSQL/reads_from_all_databases_with_the_all_keyword
test_1           | === RUN   Test_PostgreSQL/reads_two_fields_from_all_databases
test_1           | --- PASS: Test_PostgreSQL (0.31s)
test_1           |     --- PASS: Test_PostgreSQL/reads_from_one_database (0.05s)
test_1           |     --- PASS: Test_PostgreSQL/reads_from_two_databases (0.05s)
test_1           |     --- PASS: Test_PostgreSQL/reads_from_all_databases_with_the_all_keyword (0.06s)
test_1           |     --- PASS: Test_PostgreSQL/reads_two_fields_from_all_databases (0.06s)
test_1           | === RUN   Test_Mix_Mysql_PostgreSQL
test_1           | 2018/10/25 20:12:04 MySQL ready after 1/30 tries
test_1           | 2018/10/25 20:12:04 PostgreSQL ready after 1/30 tries
test_1           | === RUN   Test_Mix_Mysql_PostgreSQL/reads_from_two_databases
test_1           | === RUN   Test_Mix_Mysql_PostgreSQL/reads_from_all_databases_with_the_all_keyword
test_1           | === RUN   Test_Mix_Mysql_PostgreSQL/reads_two_fields_from_all_databases
test_1           | --- PASS: Test_Mix_Mysql_PostgreSQL (0.23s)
test_1           |     --- PASS: Test_Mix_Mysql_PostgreSQL/reads_from_two_databases (0.05s)
test_1           |     --- PASS: Test_Mix_Mysql_PostgreSQL/reads_from_all_databases_with_the_all_keyword (0.06s)
test_1           |     --- PASS: Test_Mix_Mysql_PostgreSQL/reads_two_fields_from_all_databases (0.06s)
test_1           | === RUN   TestTrimEmpty
test_1           | === RUN   TestTrimEmpty/Empty_case
test_1           | === RUN   TestTrimEmpty/Removes_multiple_empty_from_beginning,_middle_and_end
test_1           | === RUN   TestTrimEmpty/Doesn't_remove_spaces
test_1           | --- PASS: TestTrimEmpty (0.00s)
test_1           |     --- PASS: TestTrimEmpty/Empty_case (0.00s)
test_1           |     --- PASS: TestTrimEmpty/Removes_multiple_empty_from_beginning,_middle_and_end (0.00s)
test_1           |     --- PASS: TestTrimEmpty/Doesn't_remove_spaces (0.00s)
test_1           | === RUN   TestReadQuery
test_1           | === RUN   TestReadQuery/Empty_case
test_1           | === RUN   TestReadQuery/One_space
test_1           | === RUN   TestReadQuery/Trims_spaces
test_1           | === RUN   TestReadQuery/Replaces_newlines_with_spaces
test_1           | === RUN   TestReadQuery/Replaces_newlines_with_spaces,_trimming_them_on_beginning_and_end
test_1           | --- PASS: TestReadQuery (0.00s)
test_1           |     --- PASS: TestReadQuery/Empty_case (0.00s)
test_1           |     --- PASS: TestReadQuery/One_space (0.00s)
test_1           |     --- PASS: TestReadQuery/Trims_spaces (0.00s)
test_1           |     --- PASS: TestReadQuery/Replaces_newlines_with_spaces (0.00s)
test_1           |     --- PASS: TestReadQuery/Replaces_newlines_with_spaces,_trimming_them_on_beginning_and_end (0.00s)
test_1           | PASS
test_1           | ok  	sql	9.699s
```

**Pull request merged**